Document Agent Telemetry enablement by default (with Agent 7.59) (#25838)
DataDog/documentation: content/en/data_security/agent.md

For more information, see the [Secrets Management][18] documentation.

## Telemetry collection

{{< site-region region="gov" >}}

> Agent on non-government sites collects environmental, performance, and feature usage information about the Datadog Agent. When the Agent detects a government site, or [ Datadog Agent FIPS Proxy][1] is used, the Agent automatically disables this telemetry collection. When such detection is impossible (such as, for example, if a proxy is being used) Agent telemetry will be emitted but immediately dropped at Datadog's intake. To avoid this data from being emitted in the first place we recommend disabling Agent telemetry explicitly by updating the `agent_telemetry` setting in the Agent configuration file, as shown in the example below.

Reviewer: ```suggestion
Agent on non-government sites collects environmental, performance, and feature usage information about the Datadog Agent. When the Agent detects a government site, or the [Datadog Agent FIPS Proxy][1] is used, the Agent automatically disables this telemetry collection. When such detection is impossible (for example, if a proxy is being used), Agent telemetry is emitted, but immediately dropped at Datadog's intake. To avoid this data from being emitted in the first place, Datadog recommends disabling Agent telemetry explicitly by updating the `agent_telemetry` setting in the Agent configuration file, as shown in the example below.
```

You can disable this telemetry collection by updating the `agent_telemetry` setting in the Agent configuration file, as shown in the example below.
{{< tabs >}}
{{% tab "datadog.yaml" %}}

```yaml
agent_telemetry:
  enabled: false
```
{{% /tab %}}
{{% tab "Environment variables" %}}

```bash
DD_AGENT_TELEMETRY_ENABLED=false
```
{{% /tab %}}
{{< /tabs >}}

**Telemetry content:**
| Metadata ([source][1]) |
| ---------------------- |
| Machine id             |
| Machine name           |
| OS                     |
| OS version             |
| Agent version          |

| Metrics ([source][2])                       | Description                                                                                       |
| ------------------------------------------- | ------------------------------------------------------------------------------------------------- |
| checks.execution_time                       | Check's execution time in milliseconds                                                            |

> | pymem.inuse                                 | Number of bytes allocated by the python interpreter                                               |

Reviewer: ```suggestion
| pymem.inuse                                 | Number of bytes allocated by the Python interpreter                                               |
```

> | pymem.alloc                                 | Total number of bytes allocated by the python interpreter since the start of the agent            |

Reviewer: ```suggestion
| pymem.alloc                                 | Total number of bytes allocated by the Python interpreter since the start of the Agent            |
```

> | api_server.request_duration_seconds         | Total number of bytes allocated by the python interpreter since the start of the agent            |

Reviewer: ```suggestion
| api_server.request_duration_seconds         | Total number of bytes allocated by the Python interpreter since the start of the Agent            |
```
Is this correct? It's the same as the field above

Author: @hestonhoffman, missed your question actually ... looking

Author: Do not know how I missed that, fixing

| logs.decoded, logs.processed                | CLI commands execution performance (if executed)                                                  |

> | logs.sender_latency                         | Http sender latency in ms                                                                         |

Reviewer: ```suggestion
| logs.sender_latency                         | HTTP sender latency in milliseconds                                                                         |
```

> | logs.bytes_missed                           | Total number of bytes lost before they could be consumed by the agent, such as after log rotation |

Reviewer: ```suggestion
| logs.bytes_missed                           | Total number of bytes lost before they could be consumed by the Agent, such as after log rotation |
```

| logs.sent                                   | Total number of sent logs                                                                         |
| logs.dropped                                | Total number of logs dropped                                                                      |

> | logs.bytes_sent                             | Total number of bytes send before encoding if any                                                 |

Reviewer: ```suggestion
| logs.bytes_sent                             | Total number of bytes send before encoding, if any                                                 |
```

> | logs.encoded_bytes_sent                     | Total number of sent bytes after encoding if any                                                  |

Reviewer: ```suggestion
| logs.encoded_bytes_sent                     | Total number of sent bytes after encoding, if any                                                  |
```

> | dogstatsd.udp_packets                       | Dogstatsd UDP packets bytes                                                                       |
> | dogstatsd.uds_packets                       | Dogstatsd UDS packets bytes                                                                       |

Reviewer: ```suggestion
| dogstatsd.udp_packets                       | DogStatsD UDP packets bytes                                                                       |
| dogstatsd.uds_packets                       | DogStatsD UDS packets bytes                                                                       |
```

| transactions.requeued                       | Transaction requeue count                                                                         |
| transactions.retries                        | Transaction retry count                                                                           |
| point.sent                                  | Total number of sent metrics                                                                      |
| point.dropped                               | Total number of dropped metrics                                                                   |
| oracle.activity_samples_count               | Number of rows fetched in measuring query activity (Number of activity samples collected)         |
| oracle.activity_latency                     | Time to retrieve query activity in milliseconds                                                   |
| oracle.statement_metrics                    | Time to retrieve database metrics in milliseconds                                                 |
| oracle.statement_plan_errors                | Number of errors in retrieving execution plans                                                    |

> | postgres.collect_relations_autodiscovery_ms | Time to collect autodiscoverty relations in ms                                                    |
> | postgres.collect_stat_autodiscovery_ms      | Time to collect autodiscovery stats in ms                                                         |

Reviewer: ```suggestion
| postgres.collect_relations_autodiscovery_ms | Time to collect autodiscoverty relations in milliseconds                                                   |
| postgres.collect_stat_autodiscovery_ms      | Time to collect Autodiscovery stats in milliseconds                                                      |
```

> | postgres.get_new_pg_stat_activity_ms        | Time to get pg_stat_activity in ms                                                                |

Reviewer: ```suggestion
| postgres.get_new_pg_stat_activity_ms        | Time to get `pg_stat_activity` in milliseconds                                                                |
```

> | postgres.get_new_pg_stat_activity_count     | Total rows fetched to collect pg_stat_activity                                                    |

Reviewer: ```suggestion
| postgres.get_new_pg_stat_activity_count     | Total rows fetched to collect `pg_stat_activity`                                                   |
```

> | postgres.get_active_connections_ms          | Time to get active connections in ms                                                              |

Reviewer: ```suggestion
| postgres.get_active_connections_ms          | Time to get active connections in milliseconds                                                              |
```

| postgres.get_active_connections_count       | Total rows fetched to get active connections                                                      |

> | postgres.collect_activity_snapshot_ms       | Time to get activity snapshot in ms                                                               |
> | postgres.collect_statement_samples_ms       | Time to get statement samples in ms                                                               |

Reviewer: ```suggestion
| postgres.collect_activity_snapshot_ms       | Time to get activity snapshot in milliseconds                                                               |
| postgres.collect_statement_samples_ms       | Time to get statement samples in milliseconds                                                              |
```

| postgres.collect_statement_samples_count    | Total rows fetched to collect statement samples                                                   |

> Only applicable metrics are emitted. For example, if DBM is not enabled none of the database related metrics are emitted.

Reviewer: ```suggestion
Only applicable metrics are emitted. For example, if DBM is not enabled, none of the database related metrics are emitted.
```

For more information, see the [Secrets Management][18] documentation.

## Telemetry collection

{{< site-region region="gov" >}}

> Datadog on non-government sites will collect environmental, performance, and feature usage information about the Datadog Agent. When Agent detects a government site or FIPS usage Agent will automatically disable this telemetry collection. When such detection is impossible (for example when proxy is used) Agent telemetry will be emitted but immediately dropped at intake. To avoid this emission in the first place we recommend to disable Agent telemetry explicitly by updating the `agent_telemetry` setting in the Agent configuration file, as shown in the example below.

Reviewer 2: Would be useful to also mention the corresponding env var as an option to disable telemetry (`DD_AGENT_TELEMETRY_ENABLED=false` I guess)

Author: Wow, I had no idea that this would automatically work. I test it and it did. Thank you.

Author: done

{{< tabs >}}
{{% tab "datadog.yaml" %}}

```yaml
agent_telemetry:
  enabled: false
```
{{% /tab %}}
{{< /tabs >}}
{{< /site-region >}}
{{< site-region region="us,us3,us5,eu,ap1" >}}
Datadog may collect environmental, performance, and feature usage information about the Datadog Agent. This may include diagnostic logs and crash dumps of the Datadog Agent with obfuscated stack traces to support and further improve the Datadog Agent.

> You can disable this telemetry collection by updating the `agent_telemetry` setting in the Agent configuration file, as shown in the example below.

Reviewer 2: ditto: I recommend mentioning the env var as well

Author: definetly

Author: done

{{< tabs >}}
{{% tab "datadog.yaml" %}}

```yaml
agent_telemetry:
  enabled: false
```
{{% /tab %}}
{{< /tabs >}}

> **Telemetry content:**

Reviewer 2: As discussed live, please add a one-line description about what each metadata and metric represents

Author: in process

`Metadata` ([source code][1])
- Machine id
- Machine name
- OS
- OS version
- Agent version

`Metrics` ([source code][2])
- checks.execution_time
- pymem.inuse, pymem.alloc
- api_server.request_duration_seconds
- logs.decoded, logs.processed
- logs.sender_latency, logs.bytes_missed
- logs.sent, logs.dropped
- logs.bytes_sent, logs.encoded_bytes_sent
- dogstatsd.udp_packets
- dogstatsd.uds_packets
- transactions.input_count, transactions.requeued
- transactions.retries
- point.sent, point.dropped
- oracle.activity_samples_count, oracle.activity_latency
- oracle.statement_metrics, oracle.statement_plan_errors
- postgres.collect_relations_autodiscovery_ms
- postgres.collect_stat_autodiscovery_ms
- postgres.get_new_pg_stat_activity_ms
- postgres.get_new_pg_stat_activity_count
- postgres.get_active_connections_ms
- postgres.get_active_connections_count
- postgres.collect_activity_snapshot_ms
- postgres.collect_statement_samples_ms
- postgres.collect_statement_samples_count

[1]: https://github.com/DataDog/datadog-agent/blob/main/comp/core/agenttelemetry/impl/sender.go#L217-L221

> [2]: https://github.com/DataDog/datadog-agent/blob/main/comp/core/agenttelemetry/impl/config.go#L156

Reviewer 2: Recommendation for these links: link to specific commit hash or tag, so that the link remains valid even after code changes

Author: Hmm. did not know about this possibility (but I did feel that line pointing is not good enough). Thank you.

Author: done
